File: api/examples/unity-metadata-beforesend.js
// JavaScript beforeSend example
// Cleans up Unity/Android metadata pollution from exception messages

(event, hint) => {
  if (event.exception && event.exception.values) {
    for (const exception of event.exception.values) {
      // Check if metadata is polluting the exception value
      if (exception.value &&
          (exception.value.includes('Unity version') ||
           exception.value.includes('Device model') ||
           exception.value.includes('Device fingerprint'))) {

        // Extract actual exception type using regex
        const match = exception.value.match(/([\w\.]+(?:Exception|Error))/);

        if (match) {
          // Found the real exception - use it
          exception.type = match[1];
          exception.value = match[1];
        } else {
          // No exception found - use generic but descriptive title
          exception.type = 'NativeCrash';
          exception.value = 'Android Native Crash';
        }
      }
    }
  }

  return event;
}

File: api/examples/unity-metadata-beforesend.py
# Python beforeSend example
# Cleans up Unity/Android metadata pollution from exception messages

import re

def before_send(event, hint):
    if 'exception' in event and 'values' in event['exception']:
        for exception in event['exception']['values']:
            # Check if metadata is polluting the exception value
            if (exception.get('value') and
                ('Unity version' in exception['value'] or
                 'Device model' in exception['value'] or
                 'Device fingerprint' in exception['value'])):

                # Extract actual exception type using regex
                match = re.search(r'([\w\.]+(?:Exception|Error))', exception['value'])

                if match:
                    # Found the real exception - use it
                    exception['type'] = match.group(1)
                    exception['value'] = match.group(1)
                else:
                    # No exception found - use generic but descriptive title
                    exception['type'] = 'NativeCrash'
                    exception['value'] = 'Android Native Crash'

    return event
